### ISSUE TITLE
Admin Tool - report not being generated

### ISSUE THREAD

**Issue by szwart** [NASA-Tournament-Lab/NTL-ISS-Food-Intake-Tracker#191]
We're being told that the ground team is able to successfully log into the Admin Tool, but they are unable to create a report.  After clicking RUN, the page attempted to load for some time.  The ground team attempted closing the application and logging back in several times.  They even disconnected from the VNC session and tried logging back in.  They could log back in, but could not get the report to be generated.  They indicated "this is likely indicative of another problem other than the 'node.js' issue. After letting the page load for a LONG time, the ground team received the following error (photo attached)."
![iss fit proxy error - gmt 072](https://cloud.githubusercontent.com/assets/11872594/23906477/0e0245cc-089c-11e7-9fc7-c5ef17713584.png)

I'm not sure if this is an issue on NASA's side or with the Admin Tool...but I wanted to pass this along to see if you had any suggestions.

**Comment by rsial2** [comment]
@szwart was the new code deployed? We need to know what version of the system NASA is currently running.

**Comment by szwart** [comment]
No - we are many months away from that being deployed b/c of the testing it needs to go through and the availability of the lab and service pack when that will be installed.  We're still on the original version.

**Comment by rsial2** [comment]
Cheryl mentioned that they were deploying updates? Can you ask her to confirm if they've done anything? I'm worried the admin tool was updated, but the app on the ISS was not.

**Comment by coderReview** [comment]
@szwart, can you get the server logs?

**Comment by szwart** [comment]
Cheryl says "I haven't made any changes to the admin tool. It is still the version for 1.7. The only change that I can see that would affect the report would be from the restart of the server and the increase in the size of the database field. The size may be affecting the reports data. That would be the only change for ISS FIT."   
There is a meeting tomorrow morning including a large number of folks from the ground team to discuss the issue and I can see that they have requested onboard logs or other data.  I will let you know if we get any logs, but at this point we do not.

**Comment by szwart** [comment]
From the ground team "I still think there’s an underlying problem with the node.js server that is preventing BME from generating the reports, but without logging enabled on the node.js server there isn’t much that can be done. Perhaps we can try accessing the node.js server directly, without using the Apache reverse proxy but that may require modification of firewall rules on LS1 to allow that to happen."

**Comment by coderReview** [comment]
@szwart I could not reproduce the issue with the data provided from your test database. Can you check if it's a specific user that is causing the problem?
I would need the logs to find out what is going on.

**Comment by szwart** [comment]
Yes, it appears that it's a specific user. They were able to pull reports from 2 crew members but not one. We have not seen any logs available unfortunately.

On Mar 19, 2017, at 6:43 PM, Paulo Vitor Magacho <notifications@github.com<mailto:notifications@github.com>> wrote:


@szwart<https://github.com/szwart> I could not reproduce the issue with the data provided from your test database. Can you check if it's a specific user that is causing the problem?
I would need the logs to find out what is going on.

**Comment by coderReview** [comment]
@szwart is the report file for this specific user that is failing big?

**Comment by szwart** [comment]
My understanding that they are not able to pull a report at all for that user, so we do not know how big that report is.

**Comment by coderReview** [comment]
@cflottorp has said that they were able to pull the report if connecting to the admin tool without apache proxy. That is why I asked. If that's true it might be some apache configuration issue.

**Comment by cflottorp** [comment]
It seems to be specific to one user.  The user was originally entered using the “New User” route to register a new user instead of having the user included in the user profile file.  The user was able to access ISS FIT on the iPad but unfortunately it produced connectivity issues that “locked” out the other users ability to connect to the server.  The issue was corrected by increasing the size of the database for the “data”.  Now we are unable to produce the report for that user.  If “All User’s” report is used, then a proxy error is produced, if the “one user” at a time is used, then a report is generated for each of the users except the “New User” .  This is the first time the “New User” option has been used instead of the “user profile” file.

**Comment by cflottorp** [comment]
Paulo,

I am looking a the file that was dumped from the pl_fit db and the only items that I see that are inconsistent in the file is provided in the attached file.  This long string of data is between consumption records.  Without seeing a file previously, i
[ISS_FIT_DB_IssueQuest.docx](https://github.com/NASA-Tournament-Lab/NTL-ISS-Food-Intake-Tracker/files/856219/ISS_FIT_DB_IssueQuest.docx)
s this a possible issue with the database or just a dump of records?

**Comment by coderReview** [comment]
@cflottorp how did you get the data from the Word file?

@szwart , @cflottorp can you make a backup of the database from production (you can use the command below)?

`pg_dump -h 172.31.29.196 -p 56283 -U pl_fit_db -W -Fc -f pl_fit.bak pl_fit`

Change the IP  to production values.

If that is not possible (can't be done or you can't send it to me), you can run the following command in the database:

`pl_fit=# DELETE FROM data WHERE name = 'FoodProductFilter' AND value LIKE '%"removed":1%';`

This should take rid of the long line in the database. It's a good idea to do a backup before running such command.

Without the real database I can't guarantee that the command above will work.

**Comment by cflottorp** [comment]
The data is a sql file from the pl_fit.sql database file on ISS.  There are a few instances with the long string of database items (three that I have seen so far), but there are quite a few “removed” items in the file.  I will need to know the impact to the database for removing these items from the database.  Are these items that are duplicate items, issue in the database where the items are not being fully removed, items that have not been reported using the “reports”? Can you tell from the portion of the data or will you need more information in the file?

**Comment by coderReview** [comment]
Can you download the reports for this user when accessing the admin tool directly bypassing the apache proxy?
Can you send me the lines that have this huge strings?

**Comment by cflottorp** [comment]
No, we can not access the user information.  Using the ADmin Tool creates a "proxy error", which we believe is a time out error or unexpected response error.  No report is generated at all for the user.

**Comment by cflottorp** [comment]
[ISS_FIT_DB_Issue2Quest.docx](https://github.com/NASA-Tournament-Lab/NTL-ISS-Food-Intake-Tracker/files/863176/ISS_FIT_DB_Issue2Quest.docx)
[ISS_FIT_DB_NewUserLock.docx](https://github.com/NASA-Tournament-Lab/NTL-ISS-Food-Intake-Tracker/files/863177/ISS_FIT_DB_NewUserLock.docx)

Attached is another instance and data for when the "New User" was created.  When the user was created it was created using the "New User".  This may have caused issues or was a contributing factor to the connectivity issue when a value in the database was not able to be loaded. Increased the database value size and was able to connect to database but then no report was able to be created. (proxy error or no data was created.)

**Comment by coderReview** [comment]
@cflottorp got you email with the SQL file. I will load this to our VM database and will see if I can reproduce the issue. Thanks

**Comment by coderReview** [comment]
Copy this file inside "Admin Tool" folder in source directory in the server machine.

[generateSummary.py.zip](https://github.com/NASA-Tournament-Lab/NTL-ISS-Food-Intake-Tracker/files/873727/generateSummary.py.zip)

**Comment by rsial2** [comment]
Hey @cflottorp, were you able to run this? Please let us know how it goes?

**Comment by cflottorp** [comment]
Hi Rashid,

I will be able to load and test on the server this Friday. I will let you know late afternoon on Friday.

Thanks
Cheryl

Sent from my iPhone

On Mar 28, 2017, at 11:14 AM, Rashid Sial <notifications@github.com<mailto:notifications@github.com>> wrote:


Hey @cflottorp<https://github.com/cflottorp>, were you able to run this? Please let us know how it goes?

**Comment by rsial2** [comment]
hi @szwart is this error still occurring?

**Comment by szwart** [comment]
The ground team normally pulls reports for all crew once a week and delivers them to our lab.  For the past 2 weeks, each crewmember only had 1-2 items in each report.  When the crewmembers were asked if they've been recording all intake or just those few items, they indicated that they've been logging everything.  We are clearly not getting everything in the reports.  I'm told that the plan (today?) is to delete the one user who used special characters when entering food items and the ground team will then try pulling another report for the other crewmembers.

**Comment by rsial2** [comment]
Thanks, let us know how it goes.

**Comment by cflottorp** [comment]
An event also occurred just prior to the report generations that may be a contributing factor.  The server crashed and the pl_fit dbase was not able to be recovered.

**Comment by rsial2** [comment]
That could potentially be the issue. If that is the case, then it's got nothing to do with our code :)

So let's wait and see.

**Comment by rsial2** [comment]
@szwart how did this go?

**Comment by szwart** [comment]
The user profile from the user who used special keyboard characters was deleted (the crewmember had to take screenshots of the data before that could be done).  We have not been told yet whether a new report has been pulled.  Interestingly, we did notice that the few food items that were in the report pulled Monday were foods not in the food database.  They were all manually entered foods.  It was suggested to reload the food database as well.  We have not heard the outcome or if that has happened yet.

**Comment by cflottorp** [comment]
The food database was imported but only manually entered food items are still only being reported. No change in the reports generated. Only water and adhoc food items are being reported.

**Comment by rsial2** [comment]
Let's check this after the new code is deployed. Does that work for you @cflottorp?

**Comment by szwart** [comment]
Because of other priorities with the group that allows the ground testing to be completed, the service pack that will launch the new code is not slated to go up until September.  Cheryl asked if there's any way we can get logs of what is happening.  

My gut says we should have the crew reload the app and maybe that will fix this issue, realizing that we will lose the past few weeks of data.

**Comment by rsial2** [comment]
@coderReview thoughts?

**Comment by coderReview** [comment]
@szwart can I test with the database dump you sent me?

**Comment by szwart** [comment]
Sure - any advice is greatly appreciated.  The ground team now says they will not wipe and reload the app unless they can get confirmation that it will fix the problem.  The problem is difficult to recreate because there were a few steps involved (the one crewmember using special characters and then the server crash without a backup also happened).

**Comment by coderReview** [comment]
When the server crashed how did you get the data back? Should

**Comment by szwart** [comment]
We haven't gotten any data back (other than manually entered food items and Drinking Water) since the server crashed.  The crew can see their daily totals and the app is working fine for them, but we are not able to pull reports of the data on the ground.  The ground team reloaded the data.csv file and user files last week but that didn't seem to help.

**Comment by coderReview** [comment]
Ok. Understood

**Comment by szwart** [comment]
At this point, would you recommend wiping and reloading the app?  The ground folks are asking us what our forward plan is (with good rationale) and I'm afraid if we don't tell them anything, they will just stop using it.

**Comment by coderReview** [comment]
I am checking if I can force the devices to send their data again

**Comment by coderReview** [comment]
But since you reloaded the databasethe relationship between the food in the iPads and the food in the database was lost

**Comment by szwart** [comment]
Yes, it seems it was lost when the server went down and wasn't restored when we reloaded the food database.

**Comment by coderReview** [comment]
If you have an old backup it would be better than reloading the food data. If you don't have this backup I am afraid you will need to delete the app from the devices.

**Comment by coderReview** [comment]
You can make screenshots of the missing records and load them again

**Comment by szwart** [comment]
That would be too much crew time unfortunately and they will not let us do that.  It sounds like the best plan is to reload.

**Comment by coderReview** [comment]
What would be too much time? How would you reload?

**Comment by szwart** [comment]
They're willing to let us have the crew reload if that will fix the current problem.

**Comment by rsial2** [comment]
@coderReview would that fix the problem? please confirm

**Comment by szwart** [comment]
The crewmember offered to take screen shots of all data from the past month and it was on their task list today to reload the app on the iPads.  We likely won't know if it worked until Monday (when they go to pull the report).  Fingers crossed!!

**Comment by coderReview** [comment]
@rsial2 The issue now is that the local database in the iPad devices is no longer consistent with the postgresql database.  Reinstalling the app will fix this problem but it means that all the system will start from scratch.

**Comment by rsial2** [comment]
@szwart how did it go?

**Comment by rsial2** [comment]
@szwart and @cflottorp can you provide an update on this?
